airflow/contrib/hooks/gcp_vision_hook.py

```diff
@@ -0,0 +1,254 @@
+            if not explicit_name:
+                product_set.name = template_name
+            elif explicit_name != template_name:
+                raise AirflowException("The ProductSet name provided in the object ({}) is different than the"
+                                       " name created from the input parameters ({}). Please either: "
+                                       "1) Remove the ProductSet name, 2) Remove the location and "
+                                       "productset_id parameters, 3) Unify the ProductSet name and input "
+                                       "parameters.".format(explicit_name, template_name))
+        return product_set
+
+    @staticmethod
+    def _determine_product_name(product, product_id, location, project_id):
+        explicit_name = getattr(product, 'name')
+        if not location or not product_id:
+            # Not enough parameters to construct the name. Trying to use the name from Product.
+            if not explicit_name:
+                raise AirflowException("Unable to determine the Product name. Please either set the name "
+                                       "directly in the Product object or provide the `location` and "
+                                       "`product_id` parameters.")
+        else:
+            template_name = PRODUCT_NAME_TEMPLATE.format(project_id, location, product_id)
+            if not explicit_name:
+                product.name = template_name
+            elif explicit_name != template_name:
+                raise AirflowException("The Product name provided in the object ({}) is different than the"
+                                       " name created from the input parameters ({}). Please either: "
+                                       "1) Remove the Product name, 2) Remove the location and "
+                                       "product_id parameters, 3) Unify the Product name and input "
+                                       "parameters.".format(explicit_name, template_name))
+        return product
+
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def create_product(self, location, product, project_id=None,
+                       product_id=None, retry=None, timeout=None, metadata=None):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionProductCreateOperator`
+        """
+        client = self.get_client()
+        parent = PARENT_TEMPLATE.format(project_id, location)
+        self.log.info('Creating a new Product under the parent: %s', parent)
+        response = self._handle_request(lambda **kwargs: client.create_product(**kwargs), parent=parent,
+                                        product=product, product_id=product_id, retry=retry,
+                                        timeout=timeout, metadata=metadata)
+        self.log.info('Product created:\n%s', response)
```

Comment:
The whole response should be at a lower level.

Reply:
Done